crates/ark/src/variables/variable.rs

```diff
@@ -1810,4 +1857,124 @@ mod tests {
             });
         })
     }
+
+    #[test]
+    fn test_truncation() {
+        r_task(|| {
+            let vars = inspect_from_expr("as.list(1:10000)");
+            assert_eq!(vars.len(), MAX_DISPLAY_VALUE_ENTRIES);
+
+            let vars = inspect_from_expr("1:10000");
+            assert_eq!(vars.len(), MAX_DISPLAY_VALUE_ENTRIES);
+
+            let vars = inspect_from_expr("rep(letters, length.out = 10000)");
+            assert_eq!(vars.len(), MAX_DISPLAY_VALUE_ENTRIES);
+
+            let vars = inspect_from_expr("matrix(0, ncol = 10000, nrow = 10000)");
+            assert_eq!(vars.len(), MAX_DISPLAY_VALUE_ENTRIES);
+            assert_eq!(vars[0].display_value.len(), MAX_DISPLAY_VALUE_LENGTH);
+            assert_eq!(vars[0].is_truncated, true);
+
+            let vars = inspect_from_expr("new.env(parent=emptyenv())");
+            assert_eq!(vars.len(), 0);
+
+            let vars = inspect_from_expr(
+                "list2env(structure(as.list(1:10000), names = paste0('a', 1:10000)))",
+            );
+            assert_eq!(vars.len(), MAX_DISPLAY_VALUE_ENTRIES);
+            assert_eq!(vars[0].display_name, "a1");
+
+            let vars = inspect_from_expr(
+                "rep(paste0(rep(letters, length.out = 10000), collapse = ''), 10)",
+            );
+            assert_eq!(vars.len(), 10);
+            assert_eq!(vars[0].display_value.len(), MAX_DISPLAY_VALUE_LENGTH);
+            assert_eq!(vars[0].is_truncated, true);
+
+            let vars = inspect_from_expr(
+                "structure(1:10, names = rep(paste(rep(letters, length.out = 10000), collapse = ''), 10))",
+            );
+            assert_eq!(vars[0].display_name.len(), MAX_DISPLAY_VALUE_LENGTH);
+
+            let vars = inspect_from_expr(
+                "structure(as.list(1:10), names = rep(paste(rep(letters, length.out = 10000), collapse = ''), 10))",
+            );
+            assert_eq!(vars[0].display_name.len(), MAX_DISPLAY_VALUE_LENGTH);
+        })
+    }
+
+    #[test]
+    fn test_truncation_on_matrices() {
+        r_task(|| {
+            let env = harp::parse_eval_global("new.env()").unwrap();
+            harp::parse_eval0(
```

Comment:
Here as well (and below): `Environment::bind()`

Reply:
Done in https://github.com/posit-dev/ark/pull/629/commits/15b03feaafde641ff64efa9b2097d6451882c55f